File: bannerlayout/src/main/java/com/bannerlayout/Interface/OnBannerClickListener.java
package com.bannerlayout.Interface;

import android.view.View;

/**
 * by y on 2016/11/11
 * <p>
 * Banner Click event, object for the return of data
 */

public interface OnBannerClickListener<T> {
    void onBannerClick(View view, int position, T model);
}

File: bannerlayout/src/main/java/com/bannerlayout/listener/OnBannerClickListener.kt
package com.bannerlayout.listener

import android.view.View

/**
 * by y on 2016/11/11
 *
 *
 * Banner Click event, object for the return of data
 */

interface OnBannerClickListener<in T : BannerModelCallBack> {
    fun onBannerClick(view: View, position: Int, model: T)
}
